From 429ab036e5b0de2292a4b2c207acde3bd436092a Mon Sep 17 00:00:00 2001
From: Marek Szyprowski <m.szyprowski@samsung.com>
Date: Tue, 4 Sep 2012 10:44:49 +0200
Subject: [PATCH 0004/1302] fs: jbd2: force commit to free jbd2's cma/isolated
 page

Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 fs/jbd2/transaction.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/fs/jbd2/transaction.c b/fs/jbd2/transaction.c
index e0c0bc2..ca188f5 100644
--- a/fs/jbd2/transaction.c
+++ b/fs/jbd2/transaction.c
@@ -1776,8 +1776,22 @@ int jbd2_journal_try_to_free_buffers(journal_t *journal,
 		__journal_try_to_free_buffer(journal, bh);
 		jbd2_journal_put_journal_head(jh);
 		jbd_unlock_bh_state(bh);
+
+#ifndef CONFIG_CMA
 		if (buffer_jbd(bh))
 			goto busy;
+#else
+		if (buffer_jbd(bh)) {
+			unsigned mt = get_pageblock_migratetype(page);
+			/*
+			 * Workaround: In case of CMA page, just commit journal.
+			*/
+			if (mt == MIGRATE_ISOLATE || mt == MIGRATE_CMA)
+				jbd2_journal_force_commit(journal);
+			else
+				goto busy;
+		}
+#endif
 	} while ((bh = bh->b_this_page) != head);
 
 	ret = try_to_free_buffers(page);
-- 
1.8.3.2

