From 5c32e280518b5364cf35824a8f7352ebaeb5c505 Mon Sep 17 00:00:00 2001
From: Sylwester Nawrocki <s.nawrocki@samsung.com>
Date: Thu, 21 Feb 2013 15:54:19 +0100
Subject: [PATCH 0139/1302] regulator: wm8994: Enable device tree based driver
 matching

Signed-off-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/regulator/wm8994-regulator.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/regulator/wm8994-regulator.c b/drivers/regulator/wm8994-regulator.c
index a612c35..dab41ae 100644
--- a/drivers/regulator/wm8994-regulator.c
+++ b/drivers/regulator/wm8994-regulator.c
@@ -192,10 +192,17 @@ static int wm8994_ldo_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static const struct of_device_id wm8994_ldo_of_match[] = {
+	{ .compatible = "wlf,wm8994-ldo" },
+	{ },
+};
+MODULE_DEVICE_TABLE(of, wm8994_ldo_of_match);
+
 static struct platform_driver wm8994_ldo_driver = {
 	.probe = wm8994_ldo_probe,
 	.remove = wm8994_ldo_remove,
 	.driver		= {
+		.of_match_table = wm8994_ldo_of_match,
 		.name	= "wm8994-ldo",
 		.owner	= THIS_MODULE,
 	},
-- 
1.8.3.2

