From 4c5424e2cf8c623b2682679b362a418d37ba6a76 Mon Sep 17 00:00:00 2001
From: Inki Dae <inki.dae@samsung.com>
Date: Mon, 22 Jul 2013 12:47:39 +0900
Subject: [PATCH 0428/1302] dmabuf-sync: remove unnecessary the use of mutex
 lock.

Signed-off-by: Inki Dae <inki.dae@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/base/dmabuf-sync.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/base/dmabuf-sync.c b/drivers/base/dmabuf-sync.c
index 14c0c6f..fdf3bce 100644
--- a/drivers/base/dmabuf-sync.c
+++ b/drivers/base/dmabuf-sync.c
@@ -394,19 +394,16 @@ static int dmabuf_sync_get_obj(struct dmabuf_sync *sync, struct dma_buf *dmabuf,
 		return -ENOMEM;
 	}
 
+	get_dma_buf(dmabuf);
+
 	sobj->dmabuf = dmabuf;
 	sobj->robj = dmabuf->sync;
+	sobj->access_type = type;
 
 	mutex_lock(&sync->lock);
 	list_add_tail(&sobj->head, &sync->syncs);
 	mutex_unlock(&sync->lock);
 
-	get_dma_buf(dmabuf);
-
-	mutex_lock(&sobj->robj->lock);
-	sobj->access_type = type;
-	mutex_unlock(&sobj->robj->lock);
-
 	return 0;
 }
 
-- 
1.8.3.2

