From 682061cb597d5174de6fee211a2ad565c33680f7 Mon Sep 17 00:00:00 2001
From: Jonghwa Lee <jonghwa3.lee@samsung.com>
Date: Mon, 7 Oct 2013 15:45:12 +0900
Subject: [PATCH 0831/1302] WORKAROUND: Temporary workaround for
 Suspend-To-Ram.

*** Should be purged later.

Current issue :
	@m0, When system wakes from suspend-to-ram state, it often
	hangs out or shows data abort error due to invalidate PC value
	(For more information, please reference 'BSP ISSUES'
	on TizenKernel Wiki (http://10.252.81.130/mediawiki/)

Suspect :
	Non boot CPUs' hotplugging. Without power-off of non-boot cpus,
	it never happens.

Workaround :
	Keep VDD_ARM regulator ON, even system's in STR state.

Signed-off-by: Jonghwa Lee <jonghwa3.lee@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 arch/arm/boot/dts/exynos4412-slp_pq.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/exynos4412-slp_pq.dts b/arch/arm/boot/dts/exynos4412-slp_pq.dts
index 7b912dd..3021f27 100644
--- a/arch/arm/boot/dts/exynos4412-slp_pq.dts
+++ b/arch/arm/boot/dts/exynos4412-slp_pq.dts
@@ -798,7 +798,7 @@
 					regulator-max-microvolt = <1500000>;
 					regulator-always-on;
 					regulator-boot-on;
-					regulator-mem-off;
+					regulator-mem-on;
 				};
 
 				buck3_reg: buck@3 {
-- 
1.8.3.2

